zjit/src/hir.rs

```diff
@@ -4440,28 +4452,45 @@ pub fn iseq_to_hir(iseq: *const rb_iseq_t) -> Result<Function, ParseError> {
     // Compute a map of PC->Block by finding jump targets
     let jit_entry_insns = jit_entry_insns(iseq);
     let BytecodeInfo { jump_targets, has_blockiseq } = compute_bytecode_info(iseq, &jit_entry_insns);
+
+    // Make all empty basic blocks. The ordering of the BBs matters as it is taken as a schedule
```

Comment:
matters for what? what happens if we change it? can we add an assert in LIR that fails if we change this? which blocks' orders matter---the CFG should be ~orderless and we should be able to have optionals jump to any label